cmd/put.go

```diff
@@ -98,20 +87,21 @@ a file to S3.`,
 	},
 }
 
-// Put provides the ability to put objects
-type Put struct {
-	Client   s3iface.S3API
+// PutObject Represents params for object input
+type PutObject struct {
 	Bucket   string
 	Key      string
 	Source   string
 	Reader   *bufio.Reader
 	ByteSize int
 }
 
-func (p *Put) putS3Cse() ([]byte, error) {
+//PutS3Cse puts encrypted objects into S3
+func PutS3Cse(p *PutObject, encryptionclient S3ClientPutAPI) ([]byte, error) {
```

Comment:
Same question here about composition

Reply:
Implementing interfaces correctly to set up unit tests will occur first, then we can build up our composition.